spec/javascripts/models/beta/story_spec.js

```diff
@@ -1074,4 +1074,213 @@ describe('Story model', function () {
+
+    noSearchScopes.forEach(scope => {
+      describe(`when scope is ${scope}`, () => {
+        it('returns falsy', () => {
+          expect(Story.isSearch(scope)).toBeFalsy();
+        });
+      });
+
+      describe(`when scope is ${storyScopes.SEARCH}`, () => {
+        it('returns truthy', () => {
+          expect(Story.isSearch(storyScopes.SEARCH)).toBeTruthy();
+        });
+      });
+    });
+  });
+
+  describe('haveHighlightButton', () => {
+    const stories = [
+      { id: 1, storyType: storyTypes.FEATURE },
+      { id: 2, storyType: storyTypes.FEATURE },
+      { id: 3, storyType: storyTypes.FEATURE },
+    ];
+
+    const noSearchScopes = [storyScopes.ALL];
+
+    stories.forEach(story => {
+      describe(`when story id is ${story.id}`, () => {
+        noSearchScopes.forEach(scope => {
+          describe(`and scope is ${scope}`, () => {
+            it('returns falsy', () => {
+              expect(Story.haveHighlightButton(stories, story, scope)).toBeFalsy();
+            });
+          });
+        });
+
+        describe(`and scope ${storyScopes.SEARCH}`, () => {
+          it('returns truthy', () => {
+            expect(Story.haveHighlightButton(stories, story, storyScopes.SEARCH)).toBeTruthy();
+          });
+        });
+      });
+    });
+
+    describe('when story id is 100', () => {
+      const fakeStory = { id: 100, storyType: storyTypes.FEATURE };
+
+      noSearchScopes.forEach(scope => {
+        describe(`and scope is ${scope}`, () => {
+          it('returns falsy', () => {
+            expect(Story.haveHighlightButton(stories, fakeStory, scope)).toBeFalsy();
+          });
+        });
+      });
+
+      describe(`and scope ${storyScopes.SEARCH}`, () => {
+        it('returns falsy', () => {
+          expect(Story.haveHighlightButton(stories, fakeStory, storyScopes.SEARCH)).toBeFalsy();
+        });
+      });
+    });
+  });
+
+  describe('haveSearch', () => {
+    describe('when have zero search stories', () => {
+      const stories = { 
+        [storyScopes.SEARCH]: [] 
+      };
+
+      it('returns falsy', () => {
+        expect(Story.haveSearch(stories)).toBeFalsy();
+      });
+    });
+
+    describe('when have more than one search story', () => {
+      const stories = {
+        [storyScopes.SEARCH]: [
+          { id: 1, storyType: storyTypes.FEATURE },
+          { id: 2, storyType: storyTypes.FEATURE },
+          { id: 3, storyType: storyTypes.FEATURE },
+        ]
+      }
+
+      it('returns truthy', () => {
+        expect(Story.haveSearch(stories)).toBeTruthy();
+      });
+    });
+  });
+
+  describe('haveStory', () => {
+    const stories = [
+      { id: 1 },
+      { id: 2 },
+      { id: 3 },
+    ];
+
+    stories.forEach(story => {
+      describe(`when story id is ${story.id}`, () => {
+        it('returns truthy', () => {
+          expect(Story.haveStory(story, stories)).toBeTruthy();
+        });
+      });
+    });
+
+    describe('when story id is 100', () => {
```

Comment:
Tests are a way of documenting the code, we should avoid test titles that use the example data in their descriptions. `Story.haveStory` won't return `false` just because the id is `100`, it will because the array doesn't have an item with this id. Can we rename this describe (and the one at line 1271) to `when story is present in the stories array` and `when story is not present in stories array`?

Reply:
Sure. I will do that!